numpy/distutils/tests/test_system_info.py

```diff
@@ -0,0 +1,208 @@
+from __future__ import division, absolute_import, print_function
+
+import os
+from tempfile import mkstemp, mkdtemp
+
+import distutils
+from numpy.testing import *
+from numpy.distutils.system_info import *
+
+def get_class(name, notfound_action=1):
+    """
+    notfound_action:
+      0 - do nothing
+      1 - display warning message
+      2 - raise error
+    """
+    cl = {'temp1': test_temp1,
+          'temp2': test_temp2
+          }.get(name.lower(), test_system_info)
+    return cl()
+
+def get_standard_file(fname):
+    """
+    Overrides the get_standard_file from system_info
```

Comment:
Ok. I will try and work around it by incorporating it to the testcase
instead, seems easier to handle the teardown in the testcase anyway.

2015-02-25 9:15 GMT+01:00 Ralf Gommers <notifications@github.com>:

> In numpy/distutils/tests/test_system_info.py
> <https://github.com/numpy/numpy/pull/5597#discussion_r25323804>:
>
> > +
> > +def get_class(name, notfound_action=1):
> > +    """
> > +    notfound_action:
> > +      0 - do nothing
> > +      1 - display warning message
> > +      2 - raise error
> > +    """
> > +    cl = {'temp1': test_temp1,
> > +          'temp2': test_temp2
> > +          }.get(name.lower(), test_system_info)
> > +    return cl()
> > +
> > +def get_standard_file(fname):
> > +    """
> > +    Overrides the get_standard_file from system_info
>
> maybe better to give this function a different name, overriding an
> imported name seems to not serve any purpose and precludes testing that
> imported function later.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numpy/numpy/pull/5597/files#r25323804>.
>



-- 
Kind regards Nick